### ISSUE TITLE
`dependency('iconv')` doesn't work on OpenBSD.

### ISSUE THREAD

**Issue by guijan** [mesonbuild/meson#14024]
**Describe the bug**
OpenBSD doesn't have iconv in libc or in the default include/library search paths, but it does have GNU iconv as a package. On all BSDs, packages aren't in the default include/library paths. Meson's `dependency('iconv')` special case can only look up iconv in libc or the include/library paths the project has, so it doesn't find iconv on OpenBSD unless the user adds packages to the include/library paths.

**To Reproduce**
Here's a simple project showcasing the issue:
[iconv-dep.zip](https://github.com/user-attachments/files/18201757/iconv-dep.zip)

The project is small, so here's the source code inline:

_iconv-dep/iconv-dep.c_:
```c
#include <iconv.h>

int
main(void)
{
	iconv_open("", "");
}
```
_iconv-dep/meson.build_:
```meson
project('iconv-dep', 'c',
  version : '0.1',
  default_options : ['warning_level=3'])

executable('iconv-dep',
           'iconv-dep.c',
           dependencies : dependency('iconv'),
           install : true)
```

**Expected behavior**
Under the hood, meson should also look for iconv in the package include/library paths.

This works:
```sh
meson setup -Dc_args=-I/usr/local/include -Dc_link_args=-L/usr/local/lib build && meson compile -C build 
```
This doesn't work, but should:
```sh
meson setup build && meson compile -C build 
```

**system parameters**
* what operating system (e.g. MacOS Catalina, Windows 10, CentOS 8.0, Ubuntu 18.04, etc.)
OpenBSD 7.6
* what `meson --version`
Meson 1.5.1. However, a cursory look at the latest master suggests the issue is also there (no code to handle iconv outside the default paths):
https://github.com/mesonbuild/meson/blob/7a4647b194f925f386aad0c8fe5f95ff20edb0c7/mesonbuild/dependencies/misc.py#L384-L403

**Comment by eli-schwartz** [comment]
> This works:
> 
> ```shell
> meson setup -Dc_args=-I/usr/local/include -Dc_link_args=-L/usr/local/lib build && meson compile -C build 
> ```

Right, so `dependency('iconv')` *does* work on OpenBSD. :)

This is similar to any other dependency in any project at all. The only thing that is at all interesting about OpenBSD is that the dependency has a hardcoded non-default install location, rather than being wherever the user *might* choose to have it installed.

I don't know much about the BSDs but I had thought I'd heard that OpenBSD ports that expect to build against software not in the base system always add those flags anyway. Is this not expected behavior to have to do this?

**Comment by guijan** [comment]
Yes, every package that depends on other packages and needs to have the package include/library paths added to its include/library paths will have it added.
